### ISSUE TITLE
Infinite recursion in mnemonic_to_entropy_seed function in mnemonics.c

### ISSUE THREAD

**Issue by quapka** [TomasMadeja/PA193_mnemonic_BugsBunny#29]
[PA193 Aller team here]

The function [`mnemonic_to_entropy_seed`](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.c#L405) can be invoked in such a way, that it produces an infinite recursion.

While we've investigated the function further it seems, that the function [`mnemonic_to_seed`](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.c#L241) should have probably be used instead of the recursive call to back to [`mnemonic_to_entropy_seed`](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.c#L405) at [this line](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.c#L419).

The function [`mnemonic_to_entropy_seed`](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.c#L405) is not mentioned in the [README.md](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny#header-mnemonicsh), however, it is exportable and documented in [`mnemonics.h`](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.h#L81), therefore anyone can get the feeling it can be used.

I'll add a new branch (in my fork) with a new test case, showing the bug (understandably, it leads to segmentation fault).

I'm not proposing any specific fix at the moment, because it can either be fixed or the function [`mnemonic_to_entropy_seed`](https://github.com/TomasMadeja/PA193_mnemonic_BugsBunny/blob/master/src/mnemonics.c#L405) might get removed altogether (it does not add any functionality it just adds a bug now).

This issue is **severe** in the sense, that it can crash the whole application, that will call the function in question. The call to the function (as you can see in the branch, that will be linked soon) is not even improper one, it is called with valid arguments.

**Comment by quapka** [comment]
The branch [`fix-issue-29`](https://github.com/quapka/PA193_mnemonic_BugsBunny/tree/fix-issue-29) contains the [following commit](https://github.com/quapka/PA193_mnemonic_BugsBunny/commit/097149d983430eacc93c1929ffbb4422f3c13e16), that triggers the infinite recursion.

**Comment by NimRo97** [comment]
Hey, thank you for the issue. It is a total bug I made because I did not
bother to try those functions. As you said, those are completely redundant
and we included them only because their combined functionality was in the
assignment. Removing them altogether is the best fix. Good find :) I hope
you find more :D

**Comment by quapka** [comment]
I've created a pull request to merge the fix into `dev` branch as them seems to be the way you work :)

**Comment by NimRo97** [comment]
Yep, that is the way. We wish a good luck on monday to your team :)

**Comment by quapka** [comment]
Cool, thanks. Same to you guys!

**Comment by AlexInsa** [comment]
Thank you very much, and good luck for you and your team on monday too :)
